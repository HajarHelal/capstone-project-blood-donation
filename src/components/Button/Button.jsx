import React from 'react';
import { PropTypes } from 'prop-types';

function Button({ text, onclick, classFlag }) {
  const classesObject = {
    // white button
    buttonWhite:
      'inline-block text-primary-100 text-base md:text-lg  text-center md:font-medium bg-white  md:mt-8 mt-4 px-8 md:px-12 lg:px-16 py-1 lg:py-2  shadow-md transition duration-500 ease-in-out transform hover:scale-95',
    // red button
    buttonRed:
      'inline-block text-white text-base md:text-lg  text-center md:font-medium bg-primary  md:mt-8 mt-4 px-8 md:px-12 lg:px-16 py-1 lg:py-2  shadow-md transition duration-500 ease-in-out transform hover:scale-95',
  };
  return (
    <button
      type="button"
      onClick={onclick}
      // if the classFlag is true it is white button if false then red button
      className={classFlag ? classesObject.buttonWhite : classesObject.buttonRed}
    >
      {text}
    </button>
  );
}

Button.propTypes = {
  text: PropTypes.string.isRequired,
  onclick: PropTypes.func.isRequired,
  classFlag: PropTypes.bool,
};
export default Button;
