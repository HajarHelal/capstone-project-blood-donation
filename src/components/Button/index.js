// Export your components this way so that you can import it later on using
// import Button from './components/Button';
// instead of
// import Button from './components/Button/Button';

export { default } from './Button';
