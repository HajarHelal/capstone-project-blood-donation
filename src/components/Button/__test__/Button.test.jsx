import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

// This is an example of a Snapshot. It will create a new snapshot which all future
// tests will use to compare to.
it('renders correctly', () => {
  const tree = renderer
    .create(<Button text="Donate" onclick={() => console.log('saleem')} classFlag />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
