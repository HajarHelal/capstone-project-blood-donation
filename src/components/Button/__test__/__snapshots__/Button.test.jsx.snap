// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="inline-block text-primary-100 text-base md:text-lg  text-center md:font-medium bg-white  md:mt-8 mt-4 px-8 md:px-12 lg:px-16 py-1 lg:py-2  shadow-md transition duration-500 ease-in-out transform hover:scale-95"
  onClick={[Function]}
  type="button"
>
  Donate
</button>
`;
